### DISCUSSION TITLE
Scale all images in directory

### DISCUSSION THREAD

**Discussion by asoymos** [libvips/libvips#2632] [Q&A] [answered]
Hi and thanks for this amazing library, it saved me for purchasing more ram. I have a very newbie question, searched but find nothing on the matter, probably can be solved with a bash script or something but I don´t know how, so thought that maybe won't hurt asking. I am doing superresolution with the average of several images using `vips bandrank "$(echo *.tif)" final.tif`, works well but sometimes there are 15 images and something 30, so my question is how I can scale all images in a directory (so far I am doing `vips resize aligned0001.tif final/aligned_rescaled0001.tif 2` for each, but need to have 15 different long aliases which don´t seem as simple as I feel it should be). Thanks in advance.

**Comment by jcupitt** [comment, marked as answer]
Hi @asoymos,

Sorry, I'm not sure I understand. Do you want to size all images in a directory to the same size?

**Reply by asoymos** [reply]
To make a superresolution image at 200% I use 15-30 photographs, first aligned with hugin then scaled by 2 with vips and finally averaged (that is a word? sorry english isn´t my first lenguage). So I need to scale by factor 2 all the images in a directory with one command if possible.

**Reply by jcupitt** [reply]
Ah OK, then you'll need a bash loop, eg.:

```
for i in *.tif; do vips resize $i $i-2.tif 2; done
```

Or if you use something like Python you can do the resize and bandrank in one operation:

```python
#!/usr/bin/python3

import sys
import pyvips

images = [pyvips.Image.new_from_file(filename, access="sequential") for filename in sys.argv[2:]]
resized = [x.resize(s) for x in images]
final = pyvips.Image.bandjoin(resized)
final.write_to_file(sys.argv[1])
```

Put that in a file called `process.py` and you can just run:

```
./process.py out.tif source/*.tif
```

And it'll scale up everything in `source/` by x2, bandjoin them all, and save as `out.tif`.

**Reply by asoymos** [reply]
Wow thanks that is what I was talking about. You introduced me to the magic of shell scripting! (had done some javascript but didn't know that I could use variables here as well). 
One last thing, I noticed that you used bandjoin in your python script instead of bandrank. Tried it but it output a heavier image (500M bandrank and 5.3G bandjoin) which opens all black in gimp. What is the difference? (I don't even know what bands are)

**Reply by jcupitt** [reply]
oop, sorry, that was a typo, use bandrank.

**Reply by asoymos** [reply]
ok thanks!
